x-pack/plugin/migrate/src/main/java/org/elasticsearch/xpack/migrate/action/CreateIndexFromSourceAction.java

```diff
@@ -72,22 +90,34 @@ public ActionRequestValidationException validate() {
             return null;
         }
 
-        public String getSourceIndex() {
+        public String sourceIndex() {
             return sourceIndex;
         }
 
-        public String getDestIndex() {
+        public String destIndex() {
             return destIndex;
         }
 
-        public Settings getSettingsOverride() {
+        public Settings settingsOverride() {
             return settingsOverride;
         }
 
-        public Map<String, Object> getMappingsOverride() {
+        public Map<String, Object> mappingsOverride() {
             return mappingsOverride;
         }
 
+        public void settingsOverride(Settings settingsOverride) {
+            this.settingsOverride = settingsOverride;
+        }
+
+        public void mappingsOverride(Map<String, Object> mappingsOverride) {
+            this.mappingsOverride = mappingsOverride;
+        }
+
+        public void fromXContent(XContentParser parser) throws IOException {
```

Comment:
It would be good to have some unit test coverage of this method (since parsers can be tricky).